### DISCUSSION TITLE
Making `client.output` required across all functions for output consistency

### DISCUSSION THREAD

**Discussion by mikedidomizio** [vercel/vercel#10815] [Ideas]
### Idea

I was tinkering with the CLI code and I found that `output` being an optional argument for some functions was kind of odd.  If I want to `output` something, I don't want to have it dependent on if the calling function passes it in or not.  It can makes things harder to debug when issues arise.
I'm wondering if there's improvements to be made by having consistent passing of the `client.output` into functions?  By having it required, it ensures consistent output for current and future function usages.  All around better DX and debugging (also removes the need for optional chaining).

### getLatestVersion

For instance `getLatestVersion` has `output` as [optional](https://github.com/vercel/vercel/blob/main/packages/cli/src/util/get-latest-version/index.ts#L13), the only place it is being used in code has [output](https://github.com/vercel/vercel/blob/main/packages/cli/src/index.ts#L754) as a defined variable.  The only place it's actually used as an optional arg is in the tests, but the types could be updated to have it required if we simple update the tests to use `MockClient` similar to `getProjectByDeployment` [tests](https://github.com/vercel/vercel/blob/main/packages/cli/test/unit/util/projects/get-project-by-deployment.test.ts#L25).

### getProjectByDeployment

`getProjectByDeployment` also has `output` as [optional](https://github.com/vercel/vercel/blob/main/packages/cli/src/util/projects/get-project-by-deployment.ts#L19).  All use cases pass it in.  On top of that the argument probably could be eliminated since the `Client` is passed as an argument as well, and `output` is a required part of client, you get `output` for free.  
The two places it's used are [here](https://github.com/vercel/vercel/blob/main/packages/cli/src/commands/promote/request-promote.ts#L27-L32) and [here](https://github.com/vercel/vercel/blob/main/packages/cli/src/commands/rollback/request-rollback.ts#L24-L30).  Tests for this function already pass in [output](https://github.com/vercel/vercel/blob/main/packages/cli/test/unit/util/projects/get-project-by-deployment.test.ts#L25).  So Types could be updated to make it [required](https://github.com/vercel/vercel/blob/main/packages/cli/src/util/projects/get-project-by-deployment.ts#L19), optionally going the extra step and eliminating output as a separate parameter altogether.

TL;DR
- Suggesting `output` shouldn't be optional inside functions where it's used.  It can make debugging harder.  Regardless, all use cases currently pass it.
- `getProjectByDeployment` could remove `output` altogether since `client` is already passed in, and `output` is part of `client`.

P.S. I put this under ideas, not a new CLI feature, but felt better here than General
